src/data/moves/move-attrs/add-type-attr.ts

```diff
@@ -39,6 +39,6 @@ export class AddTypeAttr extends MoveEffectAttr {
   }
 
   override getCondition(): MoveConditionFunc {
-    return (_user, target, _move) => !target.isTerastallized() && !target.getTypes().includes(this.type);
+    return (_user, target, _move) => !target.isTerastallized && !target.getTypes().includes(this.type);
```

Comment:
NIT: If you dont' need params after the optionals you can omit them:

```suggestion
    return (_user, target) => !target.isTerastallized && !target.getTypes().includes(this.type);
```